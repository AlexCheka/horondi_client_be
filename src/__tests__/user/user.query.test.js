/* eslint-disable no-undef */
const { gql } = require('apollo-boost');
const userService = require('../../modules/user/user.service')
const userResolver = require('../../modules/user/user.resolver')
const client = require('../../utils/apolloClient');

require('dotenv').config();


describe('querries', () => {
    test('should recive all users like in service and resolver', async () => {
        const res = await client.query({
            query: gql`
                query {       
                    getAllUsers{
                        email
                        firstName
                        lastName
                        phoneNumber
                        purchasedProducts
                        role
                        orders
                        wishlist
                        credentials{
                            source
                            tokenPass
                        }
                        registrationDate
                        address{
                            country
                            city
                            street
                            appartment
                            buildingNumber
                        }
                    }
                }
            `
        }).then(res => {
            return res;
        })

        expect(userService.getAllUsers()).resolves.toBe(res);
        expect(userResolver.userQuery.getAllUsers()).resolves.toBe(res);

    })

    test('should recive user by token like in service and resolver', async () => {
      const res = await client.query({
          query: gql`
              query {       
                getUserByToken {
                    email
                    firstName
                    lastName
                    phoneNumber
                    purchasedProducts
                    role
                    orders
                    wishlist
                    credentials{
                        source
                        tokenPass
                    }
                    registrationDate
                    address{
                        country
                        city
                        street
                        appartment
                        buildingNumber
                    }
                }
            }
          `,
          context: {
            headers: {
                token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI1ZjA4NWQ4ZWY1YmE5MjM3NmM0NWIwZjgiLCJlbWFpbCI6InRhY2prYTMzNEBnbWFpbC5jb20iLCJpYXQiOjE1OTQzODM3NzIsImV4cCI6MTU5NDM4NzM3Mn0.ROnldc3TPO_pB_ahb3mcXJWpdsg29iejuFvmET4wji4"
            }
          }
    }).then(res => res)
    .catch(err => err)
      
      expect(userResolver.userQuery.getUserByToken(null, null,{user:{
          _id: res.data.getUserByToken._id
      }})).resolves.toBe(res);
      expect(res.data.getUserByToken).toMatchSnapshot()
  })
})