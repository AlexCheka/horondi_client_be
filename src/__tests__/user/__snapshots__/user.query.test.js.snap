// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`querries should recive user by token like in service and resolver 1`] = `
Object {
  "address": null,
  "credentials": Array [
    Object {
      "source": "horondi",
      "tokenPass": "$2a$12$RrcCJx8qZJhpvs8zQHdM8e1dXXSdOMe5tHpQUdYnuEOauJRgfI7xC",
    },
  ],
  "email": "tacjka334@gmail.com",
  "firstName": "Petro",
  "lastName": "Tatsenyak",
  "orders": Array [],
  "phoneNumber": null,
  "purchasedProducts": Array [],
  "registrationDate": "1594383758913",
  "role": null,
  "wishlist": Array [],
}
`;
